Does not compile with GHC 8

This is the compile error:

```
Configuring flexible-defaults-0.0.1.1...
    Building flexible-defaults-0.0.1.1...
    Preprocessing library flexible-defaults-0.0.1.1...
    [1 of 3] Compiling Language.Haskell.TH.FlexibleDefaults.Solve ( src/Language/Haskell/TH/FlexibleDefaults/Solve.hs, .stack-work/dist/x86_64-linu$
/Cabal-1.24.0.0/build/Language/Haskell/TH/FlexibleDefaults/Solve.o )

    /tmp/stack20475/flexible-defaults-0.0.1.1/src/Language/Haskell/TH/FlexibleDefaults/Solve.hs:13:1: warning: [-Wunused-imports]
        The import of ‘Data.Monoid’ is redundant
          except perhaps to import instances from ‘Data.Monoid’
        To import instances alone, use: import Data.Monoid()
    [2 of 3] Compiling Language.Haskell.TH.FlexibleDefaults.DSL ( src/Language/Haskell/TH/FlexibleDefaults/DSL.hs, .stack-work/dist/x86_64-linux/Ca$
al-1.24.0.0/build/Language/Haskell/TH/FlexibleDefaults/DSL.o )

    /tmp/stack20475/flexible-defaults-0.0.1.1/src/Language/Haskell/TH/FlexibleDefaults/DSL.hs:4:1: warning: [-Wunused-imports]
        The import of ‘Control.Applicative’ is redundant
          except perhaps to import instances from ‘Control.Applicative’
        To import instances alone, use: import Control.Applicative()

    /tmp/stack20475/flexible-defaults-0.0.1.1/src/Language/Haskell/TH/FlexibleDefaults/DSL.hs:9:1: warning: [-Wunused-imports]
        The import of ‘Data.Monoid’ is redundant
          except perhaps to import instances from ‘Data.Monoid’
        To import instances alone, use: import Data.Monoid()
    [3 of 3] Compiling Language.Haskell.TH.FlexibleDefaults ( src/Language/Haskell/TH/FlexibleDefaults.hs, .stack-work/dist/x86_64-linux/Cabal-1.24$
0.0/build/Language/Haskell/TH/FlexibleDefaults.o )

    /tmp/stack20475/flexible-defaults-0.0.1.1/src/Language/Haskell/TH/FlexibleDefaults.hs:73:10: error:
        • The constructor ‘InstanceD’ should have 4 arguments, but has been given 3
        • In the pattern: InstanceD clsCxt cls decs
          In the pattern: [InstanceD clsCxt cls decs]
          In a case alternative:
              [InstanceD clsCxt cls decs]
                -> do { impl <- implementDefaults defs decs;
                        return [InstanceD clsCxt cls impl] }```

--- a/src/Language/Haskell/TH/FlexibleDefaults.hs
+++ b/src/Language/Haskell/TH/FlexibleDefaults.hs
@@ -5,6 +5,7 @@
 -- 1. <https://github.com/mokus0/flexible-defaults/tree/master/examples>
 --
 -- 2. <https://github.com/mokus0/random-fu/blob/master/random-source/src/Data/Random/Internal/TH.hs>
+{-# LANGUAGE CPP #-}
 module Language.Haskell.TH.FlexibleDefaults
     ( Defaults
     , scoreBy
@@ -70,9 +71,15 @@ withDefaults defs decQ = do
     dec <- decQ
     
     case dec of
+#if MIN_VERSION_template_haskell(2,11,0)
+        [InstanceD ol clsCxt cls decs] -> do
+            impl <- implementDefaults defs decs
+            return [InstanceD ol clsCxt cls impl]
+#else
         [InstanceD clsCxt cls decs] -> do
             impl <- implementDefaults defs decs
             return [InstanceD clsCxt cls impl]
+#endif
         
         _ -> fail "withDefaults: second parameter should be a single instance declaration"
 

